tests/transition.md

```diff
@@ -277,3 +281,381 @@ All test cases described in this document are beginning in a pre-Merge world and
+  <details>
+  <summary>Click for details &#9662;</summary>
+
+  * `Genesis <- ... <- TB <- P1 <- P2 <- P3`, `TB` is a valid terminal block
+  * EL starts with `TB` as the head
+  * `newPayload(P1)`
+    * EL returns `{status: VALID, latestValidHash: payload.blockHash}`
+    * EL's head points to `TB`
+  * `newPayload(P2)`
+    * EL returns `{status: VALID, latestValidHash: payload.blockHash}`
+    * EL's head points to `TB`
+  * `newPayload(P3)`
+    * EL returns `{status: VALID, latestValidHash: payload.blockHash}`
+    * EL's head points to `TB`
+  * `forkchoiceUpdated(head: P3, safe: P2, finalized: P1)`
+    * EL's head updated to `P3`
+    * `eth_getBlockByNumber(safe)` returns `P2`
+    * `eth_getBlockByNumber(finalized)` returns `P1`
+  
+  </details>
+
+* [ ] Build atop of invalid terminal block
+  <details>
+  <summary>Click for details &#9662;</summary>
+
+  * `Genesis <- ... <- INV_TB`, `INV_TB` is a *valid PoW* but an *invalid terminal* block, i.e. `INV_TB.TD < TTD`; `INV_TB.parent.TD >= TTD` can't be checked easily as EL wouldn't process such block
+  * EL starts with `INV_TB` as the head
+  * `forkchoiceUpdated(headBlockHash: TB.blockHash, payloadAttributes: mergeTransitionBlockAttributes)`
+    * EL returns `{status: INVALID, latestValidHash: 0x00..00, payloadId: null}`
+    * EL's head points to `INV_TB` if `INV_TB.TD < TTD`, and to `INV_TB.parent` if `INV_TB.parent.TD >= TTD`
+  
+  </details>
+
+* [ ] Transition on a valid chain
+  <details>
+  <summary>Click for details &#9662;</summary>
+
+  * `Genesis <- ... <- TB <- P1 <- P2 <- P3`, `TB` is a valid terminal block
+  * EL starts with `TB` as the head
+  * `newPayload(P1)`
+    * EL returns `{status: VALID, latestValidHash: payload.blockHash}`
+    * EL's head points to `TB`
+  * `forkchoiceUpdated(head: P1, safe: 0x00..00, finalized: 0x00..00)`
+    * EL returns `{status: VALID, latestValidHash: forkchoiceState.headBlockHash}`
+    * EL's head updated to `P1`
+    * `eth_getBlockByNumber(safe)` returns `-39001: Unknown block` error
+    * `eth_getBlockByNumber(finalized)` returns `-39001: Unknown block` error
+  * `newPayload(P2) + forkchoiceUpdated(head: P2, safe: P1, finalized: 0x00..00)`
+    * EL's head updated to `P2`
+    * `eth_getBlockByNumber(safe)` returns `-39001: Unknown block` error
+    * `eth_getBlockByNumber(finalized)` returns `-39001: Unknown block` error
+  * `newPayload(P3) + forkchoiceUpdated(head: P3, safe: P2, finalized: P1)`
+    * EL's head updated to `P3`
+    * `eth_getBlockByNumber(safe)` returns `P2`
+    * `eth_getBlockByNumber(finalized)` returns `P1`
+  
+  </details>
+
+* [ ] Transition on an invalid chain
+  <details>
+  <summary>Click for details &#9662;</summary>
+
+  * `Genesis <- ... <- INV_TB <- P1`, `INV_TB` is a *valid PoW* but an *invalid terminal* block:
+    * [ ] `INV_TB.TD < TTD`
+    * [ ] `INV_TB.parent.TD >= TTD` -- it might require a second EL with a higher `TTD` value
```

Comment:
I think this variant is covered by https://github.com/ethereum/hive/pull/554.
And actually `INV_TB.TD < TTD` should be technically covered too, the only difference is that the client does not start with this Block and has to sync it from another client.